geode-core/src/main/java/org/apache/geode/internal/tcp/Connection.java

```diff
@@ -1692,10 +1700,8 @@ private void readMessages() {
         }
       }
     } finally {
-      if (!isHandShakeReader) {
-        synchronized (stateLock) {
-          connectionState = STATE_IDLE;
-        }
+      synchronized (stateLock) {
```

Comment:
The (!isHandShakeReader) test was added here to fix a race condition between the handshake thread and an application thread using the thread-owned Connection.  I think this test needs to remain in place.

Reply:
thanks for comments. I will return check on !isHandShakeReader.